conda/core/python_dist.py

```diff
@@ -0,0 +1,1103 @@
+# This is useful for parsing, for example, `Python-Version`
+PARTIAL_PYPI_SPEC_PATTERN = re.compile(r'''
+    # Text needs to be stripped and all extra spaces replaced by single spaces
+    (?P<name>^[A-Z0-9][A-Z0-9._-]*)?
+    \s?
+    (\[(?P<extras>.*)\])?
+    \s?
+    (?P<constraints>\(? \s? ([\w\d<>=!~,\s\.\*]*) \s? \)? )?
+    \s?
+''', re.VERBOSE | re.IGNORECASE)
+PySpec = namedtuple('PySpec', ['name', 'extras', 'constraints', 'marker', 'url'])
+
+
+class MetadataWarning(Warning):
+    pass
+
+
+class PythonDistributionMetadata(object):
+    """
+    Object representing the metada of a Python Distribution given by anchor
+    file (or directory) path.
+
+    This metadata is extracted from a single file. Python distributions might
+    create additional files that complement this metadata information, but
+    that is handled at the python distribution level.
+
+    Notes
+    -----
+      - https://packaging.python.org/specifications/core-metadata/
+      - Metadata 2.1: https://www.python.org/dev/peps/pep-0566/
+      - Metadata 2.0: https://www.python.org/dev/peps/pep-0426/ (Withdrawn)
+      - Metadata 1.2: https://www.python.org/dev/peps/pep-0345/
+      - Metadata 1.1: https://www.python.org/dev/peps/pep-0314/
+      - Metadata 1.0: https://www.python.org/dev/peps/pep-0241/
+    """
+    FILE_NAMES = ('METADATA', 'PKG-INFO')
+
+    # Python Packages Metadata 2.1
+    # -----------------------------------------------------------------------------
+    SINGLE_USE_KEYS = frozendict([
+        ('Metadata-Version', 'metadata_version'),
+        ('Name', 'name'),
+        ('Version', 'version'),
+        ('Summary', 'summary'),
+        ('Description', 'description'),
+        ('Description-Content-Type', 'description_content_type'),
+        ('Keywords', 'keywords'),
+        ('Home-page', 'home_page'),
+        ('Download-URL', 'download_url'),
+        ('Author', 'author'),
+        ('Author-email', 'author_email'),
+        ('Maintainer', 'maintainer'),
+        ('Maintainer-email', 'maintainer_email'),
+        ('License', 'license'),
+        # Deprecated
+        ('Obsoleted-By', 'obsoleted_by'),  # Note: See 2.0
+        ('Private-Version', 'private_version'),  # Note: See 2.0
+    ])
+    MULTIPLE_USE_KEYS = frozendict([
+        ('Platform', 'platform'),
+        ('Supported-Platform', 'supported_platform'),
+        ('Classifier', 'classifier'),
+        ('Requires-Dist', 'requires_dist'),
+        ('Requires-External', 'requires_external'),
+        ('Requires-Python', 'requires_python'),
+        ('Project-URL', 'project_url'),
+        ('Provides-Extra', 'provides_extra'),
+        ('Provides-Dist', 'provides_dist'),
+        ('Obsoletes-Dist', 'obsoletes_dist'),
+        # Deprecated
+        ('Extension', 'extension'),  # Note: See 2.0
+        ('Obsoletes', 'obsoletes'),
+        ('Provides', 'provides'),
+        ('Requires', 'requires'),
+        ('Setup-Requires-Dist', 'setup_requires_dist'),  # Note: See 2.0
+    ])
```

Comment:
I did this...

```python
    # Python Packages Metadata 2.1
    # -----------------------------------------------------------------------------
    SINGLE_USE_KEYS = frozendict((
        ('Metadata-Version', 'metadata_version'),
        ('Name', 'name'),
        ('Version', 'version'),
        # ('Summary', 'summary'),
        # ('Description', 'description'),
        # ('Description-Content-Type', 'description_content_type'),
        # ('Keywords', 'keywords'),
        # ('Home-page', 'home_page'),
        # ('Download-URL', 'download_url'),
        # ('Author', 'author'),
        # ('Author-email', 'author_email'),
        # ('Maintainer', 'maintainer'),
        # ('Maintainer-email', 'maintainer_email'),
        ('License', 'license'),
        # # Deprecated
        # ('Obsoleted-By', 'obsoleted_by'),  # Note: See 2.0
        # ('Private-Version', 'private_version'),  # Note: See 2.0
    ))
    MULTIPLE_USE_KEYS = frozendict((
        ('Platform', 'platform'),
        ('Supported-Platform', 'supported_platform'),
        ('Classifier', 'classifier'),
        ('Requires-Dist', 'requires_dist'),
        ('Requires-External', 'requires_external'),
        ('Requires-Python', 'requires_python'),
        # ('Project-URL', 'project_url'),
        ('Provides-Extra', 'provides_extra'),
        # ('Provides-Dist', 'provides_dist'),
        # ('Obsoletes-Dist', 'obsoletes_dist'),
        # # Deprecated
        # ('Extension', 'extension'),  # Note: See 2.0
        # ('Obsoletes', 'obsoletes'),
        # ('Provides', 'provides'),
        ('Requires', 'requires'),
        # ('Setup-Requires-Dist', 'setup_requires_dist'),  # Note: See 2.0
    ))
```

I don't want us parsing stuff that we're never going to use.  At least right now.  I'm fine leaving them as comments though indefinitely, so if we ever need to add anything back in someday, it's clearer what's there to add.